lib/watcher.js

```diff
@@ -44,12 +44,10 @@ function Watcher(logger, api, files, sources) {
 
 	this.isTest = makeTestMatcher(files, AvaFiles.defaultExcludePatterns());
 
-	var isFirstRun = true;
 	this.clearLogOnNextRun = true;
+	this.runVector = 0;
 	this.run = function (specificFiles) {
-		if (isFirstRun) {
-			isFirstRun = false;
-		} else {
+		if (this.runVector > 0) {
```

Comment:
`this.runVector > 0` is the best option here. I would use `!== 0` when it can also be negative.